### ISSUE TITLE
details

### ISSUE THREAD

**Issue by YOKE** [liutinglt/CE2P#14]
Could you please explain the label_rev and how to get them?  And how I know if the native CUDA implementation of InPlace-ABN succeeds?Thanks for your reply!

**Comment by BEYBB7** [comment]
I can't find the label_rev. Could you help me?@liutinglt

**Comment by eng100200** [comment]
In the train dataset lable_rev are already provided

**Comment by eng100200** [comment]
Please download imagenet pretrained resent-101, label files of edge and the trained models from baidu drive or Google drive, and put it into dataset folder.

**Comment by BEYBB7** [comment]
> In the train dataset lable_rev are already provided
In the  baidu drive or Google drive, there are  imagenet pretrained resent-101, label files of edge and the trained models, I didn't find the reversed images. And in LIP, I only have train and annotation images. ??I still don't know where they are. Could you tell me more specifically? Thank you very much~~~

**Comment by BEYBB7** [comment]
> Please download imagenet pretrained resent-101, label files of edge and the trained models from baidu drive or Google drive, and put it into dataset folder.

If it‘s convenient，can I have your email address. my email: 1256798013@qq.com

**Comment by liutinglt** [comment]
@BEYBB7  Just flip the labels horizontally.  NOTE, for the flipped label, you must swap the left and right label.

**Comment by BEYBB7** [comment]
> @BEYBB7 Just flip the labels horizontally. NOTE, for the flipped label, you must swap the left and right label.

Have you provided the reversed images? I didn't find them

**Comment by liutinglt** [comment]
@BEYBB7  I haven't provided, as it's easy to generate by your own.

**Comment by BEYBB7** [comment]
> @BEYBB7 I haven't provided, as it's easy to generate by your own.

Thank you！Got it

**Comment by BEYBB7** [comment]
@liutinglt   
What's does the difference between the pretrained  model-- LIP_CE2P_train.pth, LIP_CE2P_trainVal_321_681.pth, LIP_CE2P_train_473.pth? Could you explain the mode of training of them?

**Comment by eng100200** [comment]
LIP_CE2P_train.pth...model obtained using the training images only 

LIP_CE2P_trainVal_321_681.pth  ,,,,,model obtained using the only validation images as training images 


LIP_CE2P_train_473.pth model after resizing the images to 473 x 473

**Comment by BEYBB7** [comment]
Could you tell the detail of your  baseline-resnet101 when you implement it? Especially, the part of decoder? @liutinglt

**Comment by zzw1123** [comment]
@liutinglt Dear ting, I have read your paper and it is an excellent job!
And I have a question about the imagenet pre-trained model, is it downloaded directly from deeplab_v2 project? Or have you modified it?

**Comment by liutinglt** [comment]
@BEYBB7 The baseline performance is obtained by predicting from layer4 module with 1X1 CONV directly.  Remove layer5, edgelayer , layer6, layer7 in models.py, and replace with nn.Conv2d(2048, num_classes, kernel_size=1, padding=0, dilation=1, bias=True)

**Comment by liutinglt** [comment]
@zzw1123 The pretriained model is converted from deeplab_V2, which has been pretrained on MS-COCO dataset. It's provided in https://github.com/speedinghzl/Pytorch-Deeplab and https://github.com/isht7/pytorch-deeplab-resnet

**Comment by BEYBB7** [comment]
> @BEYBB7 The baseline performance is obtained by predicting from layer4 module with 1X1 CONV directly. Remove layer5, edgelayer , layer6, layer7 in models.py, and replace with nn.Conv2d(2048, num_classes, kernel_size=1, padding=0, dilation=1, bias=True)

Is there any different setting? I‘can't implement your result -- almost 48

**Comment by zzw1123** [comment]
@BEYBB7  Me, neither. And the mIoU is only 35.47%, what is your result?

**Comment by BEYBB7** [comment]
> @BEYBB7 Me, neither. And the mIoU is only 35.47%, what is your result?

About 41%

**Comment by zzw1123** [comment]
@BEYBB7 Do you use the same parameters as those in the paper?

**Comment by BEYBB7** [comment]
> @BEYBB7 Do you use the same parameters as those in the paper?

Yes，I don't change any parameter.

**Comment by zzw1123** [comment]
@BEYBB7 That is confusing...
@liutinglt Could you please help us?

**Comment by GengDavid** [comment]
@BEYBB7 @zzw1123 I cannot reproduce the results either. BTW, May I know how many GPUs did you use?

**Comment by zzw1123** [comment]
@GengDavid When I tried to re-implement the baseline result, I used 5 gpus and the same lr as paper mentioned. How about you?

**Comment by LIDChallenge** [comment]
@YOKE @eng100200 @zzw1123 @GengDavid As there are some problems strange with Pytorch 0.3.1, please using the updated code with Pytorch 0.4.1.

**Comment by liutinglt** [comment]
@YOKE @eng100200 @zzw1123 @GengDavid  As there are some problems strange with Pytorch 0.3.1, please using the updated code with Pytorch 0.4.1.

**Comment by YOKE** [comment]
There is an error when I use your new project, could you help me?
![qq 20190227170728](https://user-images.githubusercontent.com/34132063/53479150-5fa95280-3ab3-11e9-940b-7fb74e33e17f.png)

**Comment by liutinglt** [comment]
@YOKE  The modules is deleted in this version.  Please build the libs following Readme,  and use the  'CE2P.py' in 'networks'

**Comment by YOKE** [comment]
> @YOKE The modules is deleted in this version. Please build the libs following Readme, and use the 'CE2P.py' in 'networks'

![1](https://user-images.githubusercontent.com/34132063/53484801-a0a76400-3abf-11e9-9870-f305261dde58.png)
It's the same problem. @liutinglt

**Comment by YOKE** [comment]
Is there something different and I need to check something? @liutinglt

**Comment by zzw1123** [comment]
@liutinglt My coda got stucked during forward process. And I found it stops before layer4 in resnet101. Do you know why ?

**Comment by YOKE** [comment]
> @liutinglt My code got stucked during forward process. And I found it stops before layer4 in resnet101. Do you know why ?

Have you met my error--undefined symbol PyInt_FromLong?

**Comment by liutinglt** [comment]
@YOKE Did you delete the libs/_ext, and rebuild it by yourself?

**Comment by liutinglt** [comment]
@zzw1123 First, you can try smaller batch size. 
   Or, you can use Pytorch 1.0, just download 'modules' in https://github.com/mapillary/inplace_abn, and rename it with 'libs'.

**Comment by YOKE** [comment]
> @YOKE Did you delete the libs/_ext, and rebuild it by yourself?

Yes, I've rebuilt it. However, I'am meeting the error when run `./run_evaluate.sh`. I have no idea to fix it. @liutinglt

**Comment by YOKE** [comment]
> > @YOKE Did you delete the libs/_ext, and rebuild it by yourself?
> 
> Yes, I've rebuilt it. However, I'am meeting the error when run `./run_evaluate.sh`. I have no idea to fix it. @liutinglt

python build.py  The result is 
![2](https://user-images.githubusercontent.com/34132063/53491038-495dbf80-3ad0-11e9-9aff-29403687b59b.png)
Whether the  warnings make any defference?

**Comment by zzw1123** [comment]
@liutinglt Thanks and after changing batch size from 24 to 15, it goes well.

**Comment by zzw1123** [comment]
@liutinglt Hi, me again. I found that during training process, the Utilization of GPU may be 0% sometimes, is it because of the processing of the images such as scaling and affine transform?

**Comment by 994374821** [comment]
> @liutinglt
> What's does the difference between the pretrained model-- LIP_CE2P_train.pth, LIP_CE2P_trainVal_321_681.pth, LIP_CE2P_train_473.pth? Could you explain the mode of training of them?

Hi, where did you find the pretrained models of LIP_CE2P_train.pth, LIP_CE2P_trainVal_321_681.pth, LIP_CE2P_train_473.pth ? I can not find a link to download them.

**Comment by YOKE** [comment]
The project has been updated and the link has been deleted.
